src/daft-core/src/array/ops/utf8.rs

```diff
@@ -479,6 +481,89 @@ impl Utf8Array {
         Ok(Utf8Array::from((self.name(), Box::new(arrow_result))))
     }
 
+    pub fn find(&self, substr: &Utf8Array) -> DaftResult<Int64Array> {
+        let self_arrow = self.as_arrow();
+        let substr_arrow = substr.as_arrow();
+        // Handle empty cases.
+        if self.is_empty() || substr_arrow.is_empty() {
```

Comment:
Perhaps `self.is_empty() && substr_arrow.is_empty()` instead as the fast-path for empty data?

I think `find([], ["foo"]` is actually an error, since the lengths don't match.


```suggestion
        if self.is_empty() && substr_arrow.is_empty() {
```

Reply:
all string expressions handle all_empty data or pattern case such way (returning an empty series).
If we consider this to be an error case, maybe we need to make change in other expressions too ? @jaychia

Reply:
Ah yes, I see that this is the case. cc @colin-ho as well for thoughts.

Reply:
I think we can do `self.is_empty() || substr_arrow.is_empty()` if we have a prior check for correct lengths, i.e. we first check if `self.len == other.len || (self.len == 1 && other.len > 1) || (self.len > 1 && other.len == 1)`, then we can do the empty array fast path and potentially also full null input fast path.

Reply:
To be handled via #2058 
cc : @colin-ho